### PR TITLE
Modifications to allow custom time intervals in a timelapse

### PR DESCRIPTION
**Pull request by ckolluru**
Based on the discussion in https://github.com/micro-manager/pycro-manager/issues/602#issue-1729633195

I'm assuming the documentation will be automatically updated in the next build.

### CODE REVIEW

**Comment by henrypinkard** [comment]
Looks good. you may also want to add a test here: https://github.com/micro-manager/pycro-manager/blob/main/pycromanager/test/test_MDA_events.py

**Comment by ckolluru** [comment]
Will add the tests.

Is there a need to check whether values in the list are greater than or equal to 0?
If ```min_start_time``` is less than zero, is it handled the same way as if it were zero?

**Comment by henrypinkard** [comment]
Cool looks good!

**Comment by henrypinkard** [comment]
> Is there a need to check whether values in the list are greater than or equal to 0?
> If `min_start_time` is less than zero, is it handled the same way as if it were zero?

I think it will be just treated like it was 0

> Also, which script runs the test cases?

 You can run each test script locally. They also run in github actions before PRs go through. This is what tells the action to run:
https://github.com/micro-manager/pycro-manager/blob/main/.github/workflows/build_and_test.yml

**Comment by ckolluru** [comment]
Sounds good, thanks.

File: pycromanager/test/test_MDA_events.py [hunk 1]
```diff
@@ -84,6 +84,32 @@ def test_time_points():
...
+    assert expected == multi_d_acquisition_events(num_time_points=5, time_interval_s=[0, 10, 20, 0, 10])
+
+    expected = [
+        {"axes": {"time": 0}, "min_start_time": 0},
+        {"axes": {"time": 1}, "min_start_time": 10},
+        {"axes": {"time": 2}, "min_start_time": 20},
+        {"axes": {"time": 3}, "min_start_time": 30},
+        {"axes": {"time": 4}, "min_start_time": 20},
+    ]
+    assert expected == multi_d_acquisition_events(num_time_points=5, time_interval_s=[0, 10, 10, 10, -10])
```
Comment (ieivanov) [inline pycromanager/test/test_MDA_events.py:112]: Should we require that `time_interval_s` is non-negative? pycromanager executes events in order, so here the acquisition engine will execute event `{"axes": {"time": 2}}`, wait 10 seconds, execute event `{"axes": {"time": 3}}` and then execute event `{"axes": {"time": 4}}` right away. I don't think there is ever a case where a negative `time_interval_s` will have any effect.

Comment (henrypinkard) [reply pycromanager/test/test_MDA_events.py:112]: Yes I also think that it won't cause a problem if its negative. Maybe just a warning to let the user know

**Pull request merged**